numba/_helperlib.c

```diff
@@ -978,6 +978,35 @@ numba_do_raise(PyObject *exc_packed)
     return status;
 }
 
+#ifdef NUMBA_COMPILING_AOT
+/* AOT avoid the use of `numba.core.serialize` */
+NUMBA_EXPORT_FUNC(PyObject *)
+numba_unpickle(const char *data, int n, const char *hashed)
+{
+    PyObject *buf, *obj;
+    static PyObject *loads;
+
+    /* Caching the pickle.loads function shaves a couple µs here. */
+    if (loads == NULL) {
+        PyObject *picklemod;
+        picklemod = PyImport_ImportModule("pickle");
+        if (picklemod == NULL)
+            return NULL;
+        loads = PyObject_GetAttrString(picklemod, "loads");
+        Py_DECREF(picklemod);
+        if (loads == NULL)
+            return NULL;
+    }
+
+    buf = PyBytes_FromStringAndSize(data, n);
+    if (buf == NULL)
+        return NULL;
+    obj = PyObject_CallFunctionObjArgs(loads, buf, NULL);
+    Py_DECREF(buf);
+    return obj;
+}
+
+#else
 
 NUMBA_EXPORT_FUNC(PyObject *)
```

Comment:
Does this imply that AOT compiled functions do not have the same level of e.g. closure compilation support as those which are jitted and cached?

Reply:
The only difference here is the lack of in-memory cache for the unpickled reuslts. The `_numba_unpickle()` method also relies on the same `pickle.loads()`. 

If we want to do better for AOT unpickling performance, we can implement `_numba_unpickle()` in C.